### ISSUE TITLE
Endpoints for terminated pods ratelimit clusters not removed immediately

### ISSUE THREAD

**Issue by bewebi** [solo-io/gloo#8942]
### Gloo Edge Product

Enterprise

### Gloo Edge Version

v1.16.0-beta2

### Kubernetes Version

v1.26.3

### Describe the bug

When a pod is deleted for the ratelimit upstream, the endpoint for it will only be removed after it's failed the health check

Our health check is configured with a 60s interval and unhealthy threshold of 5, so for five minutes after pod deletion, requests may time out or return `5xx` responses

### Expected Behavior

When a ratelimit pod is deleted, no traffic is ever routed to it

### Steps to reproduce the bug

This was encountered when e2e testing Gloo Portal, particularly tests added in solo-io/dev-portal#2837

To reproduce, one can check out that PR's branch*, set up a test cluster using 
```
./ci/cluster_setup/test-cluster.sh setup -gt gloo -gv v1.16.0-beta2 --feature-shared-api-docs true
```
and then run the Aerospike e2e tests, which must patch the rate-limit deployment prior to running, and therefore deletes the pre-existing rate-limit pod

Observe that the ratelimit test for shared routes fails because it makes more requests than expected for a particular key to be rate-limited
To make this clearer to observe, set `ratelimitServer.requestTimeout` >1s in the `Settings` CR

One can also retrieve an Envoy config dump within the 5 minute window after the rate-limit pod is deleted and observe that there are endpoints in the `rate-limit_gloo-portal` cluster for both the new pod's IP and the old pod's IP

To observe that the test passes when the pod is not restarted first, one can execute the patch that the test does 5+ minutes prior to running the tests so that the patch does not cause a change (or pod deletion) when executed by the test
```
kubectl patch -n gloo-portal deployments.apps rate-limit --patch '{"spec":{"template":{"spec":{"containers":[{"name":"rate-limit","env":[{"name":"AEROSPIKE_ADDRESS","value":"aerospike.aerospike.svc.cluster.local"},{"name":"AEROSPIKE_NAMESPACE","value":"aerospike"}]}]}}}}'
```

* At [this commit](https://github.com/solo-io/dev-portal/pull/2837/commits/f73e3a01a021022a1fbe0dd3fd58e585bee1e26e) the tests are setup to highlight this bug, with the relevant test focused, debug logging present, and an extended timeout of 20 minutes for the ratelimit test which allows the user to observe that requests start flowing as intended after 5 minutes

### Additional Context

This bug is a blocker for solo-io/dev-portal#2771 as the Aerospike e2e tests will not run properly until this bug is resolved

[This branch](https://github.com/solo-io/gloo/compare/deleted-pods-healthy-bug) has a one-line change to include `IgnoreHealthOnHostRemoval` on EDS clusters that seems to fix the bug, however more investigation needs to be done to determine whether this should be configurable, exactly how the fix should be implemented, whether the health checks should be modified to reduce how long unhealthy pods are considered healthy, etc.

**Comment by bewebi** [comment]
There is a PR open for this issue  (solo-io/solo-projects#5537) which has been held up due to unexplained behavior in e2e testing
While it would be ideal to root that out, a lot of time has been spent on doing so already without reaching a conclusion, and the config change itself is not particularly controversial
Therefore, in the interest of resolving this issue without devoting significant additional resources, we will essentially allow for some degree of flakiness in the tests without totally compromising them
The PR should be able to be reviewed and merged with little additional effort

**Comment by bewebi** [comment]
A fix for this has been merged to `main`, `v1.16.x`, and `v1.15.x` and will be included in the next release for each

Backporting to earlier versions was non-trivial and deemed not worth the effort
